src/main/java/ro/unibuc/hello/services/CheckUserRightsService.java

```diff
@@ -0,0 +1,51 @@
+package ro.unibuc.hello.services;
+
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import ro.unibuc.hello.dtos.ActionDTO;
+import ro.unibuc.hello.exceptions.UserNotAuthorizedException;
+
+@Component
+public class CheckUserRightsService {
+
+    @Autowired
+    private UserService userService;
+
+    @Autowired
+    private ActionService actionService;
+
+    public Boolean checkUserRights(String id, String actionCode) {
+        ActionDTO action = actionService.getActionById(actionCode);
```

Comment:
You could split these function into 3: the main one checkUserRights that calls on checkPolicies and checkRoles (this one would also call on checkPolicies as there is duplicated code) or into 2: checkUserRights that calls checkPolicies and then for all roles you call on checkPolicies(roles.getPolicies()). This is merely a suggestion and not a requirement.